website/content/docs/concepts/host-discovery/gcp.mdx

```diff
@@ -0,0 +1,225 @@
+    name        = "GCP Catalog"
+    description = "GCP Host Catalog"
+    scope_id    = boundary_scope.project.id
+    plugin_name = "gcp"
+
+    attributes_json = jsonencode({
+      "zone" = "us-central1-a ",
+      "project_id" = var.gcp_project_id,
+      "client_email" = var.gcp_client_email,
+      "disable_credential_rotation" = true })
+    secrets_json = jsonencode({
+      "private_key_id" = var.private_key_id,
+      "private_key" = var.private_key})
+  }
+  ```
+
+</Tab>
+</Tabs>
+
+### Authentication with service account impersonation
+Service account impersonation allows Boundary to authenticate with GCP using a service account that impersonates another service account. 
+This is useful when you want to restrict the permissions of the service account used by Boundary. Two service accounts are required:
+
+- `Base service account`: The service account used by Boundary to authenticate with GCP. The service account must have the following roles:
+  - `roles/iam.serviceAccountTokenCreator` (Required) - To create a token for the target service account.
+  - `roles/iam.serviceAccountKeyAdmin` (Optional) - To rotate the service account key if `disable_credential_rotation` is set to `false`.
+
+- `Target service account`: The service account to impersonate. The service account must have the following roles:
+  - `roles/compute.viewer` (Required): To list Compute Engine VM instance in the project.
+
+Pass the following fields to Boundary for authentication:
+
+<Tabs>
+<Tab heading="CLI">
+
+```shell-session
+$ boundary host-catalogs create plugin \
+  -scope-id $BOUNDARY_PROJECT_ID \
+  -plugin-name gcp \
+  -attr disable_credential_rotation=true \
+  -attr zone=us-central1-a \
+  -attr project_id=env://GCP_PROJECT_ID \
+  -attr client_email=env://BASE_SERVICE_ACCOUNT_EMAIL \
+  -attr target_service_account_id=env://TARGET_SERVICE_ACCOUNT_EMAIL \
+  -secret private_key_id=env://BASE_SERVICE_ACCOUNT_PRIVATE_KEY_ID \
+  -secret private_key=file://$BASE_SERVICE_ACCOUNT_PRIVATE_KEY_FILE_PATH
+```
+
+</Tab>
+<Tab heading="Terraform">
+
+  ```hcl
+  resource "boundary_host_catalog_plugin" "gcp_host_catalog" {
+    name        = "GCP Catalog"
+    description = "GCP Host Catalog"
+    scope_id    = boundary_scope.project.id
+    plugin_name = "gcp"
+
+    attributes_json = jsonencode({
+      "zone" = "us-central1-a ",
+      "project_id" = var.gcp_project_id,
+      "client_email" = var.gcp_base_service_account_email,
+      "target_service_account_id" = var.gcp_target_service_account_email,
+      "disable_credential_rotation" = true })
+    secrets_json = jsonencode({
+      "private_key_id" = var.base_service_account_private_key_id,
+      "private_key" = var.base_service_account_private_key})
+  }
+  ```
+
+</Tab>
+</Tabs>
+
+### Authentication with Application Default Credentials (ADC)
+If you are running Boundary on a GCP VM instance, you can use Application Default Credentials (ADC) to authenticate with GCP.
```

Comment:
Non-native English speakers struggle with "-ing" words sometimes b/c they can be verbs or nouns in English, so I made suggestions to use present tense verbs instead of present participles.